core/alsp_src/generic/bsio.c

```diff
@@ -334,7 +339,46 @@ long standard_console_error(char *buf, long n)
 
 long standard_console_read(char *buf, long n)
 {
-    return read(STDIN_FILENO, buf, n);
+    char *line;
+    long count;
+
+        /* Load the history file once (i.e., at startup) 
+    	   do_load_prev_history is set == 1 by default (set == 0 during shell startup)
+	   Setting it to 0 ensures the
+	   history will only be loaded once during an alspro shell session.
+	*/
+    if (do_load_prev_history == 1){
+        linenoiseHistoryLoad(history_file); 
+        do_load_prev_history = 0;
+    }
+
+    if (console_prompt == NULL){
+	console_prompt = "?- ";
```

Comment:
I don't think the  constant `?-` should be here. There are couple ways to handle default for `console_prompt`.

Simplest is to initialize `console_prompt` to the empty string, rather then `NULL` -- then no guard-clause would be needed in `standard_console_read` because `console_prompt` will always have a legal value.

Another option is to have the guard-clause return an error with appropriate errno setting.